[Style] Convert the internal marquee properties to strong style types

weinig opened the pull request:
#### 5a56bacea5f8e2caf229ce355689cf089372293b
<pre>
[Style] Convert the internal marquee properties to strong style types
<a href="https://bugs.webkit.org/show_bug.cgi?id=298475">https://bugs.webkit.org/show_bug.cgi?id=298475</a>

Reviewed by Darin Adler.

Converts the internal marquee properties, `-webkit-marquee-increment`,
`-webkit-marquee-repetition` and `-webkit-marquee-speed`, to strong
style types.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/html/HTMLMarqueeElement.cpp:
* Source/WebCore/rendering/RenderMarquee.cpp:
* Source/WebCore/rendering/RenderMarquee.h:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleMarqueeData.cpp:
* Source/WebCore/rendering/style/StyleMarqueeData.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/values/non-standard/StyleWebKitMarqueeIncrement.h: Added.
* Source/WebCore/style/values/non-standard/StyleWebKitMarqueeRepetition.h: Added.
* Source/WebCore/style/values/non-standard/StyleWebKitMarqueeSpeed.h: Added.

Canonical link: <a href="https://commits.webkit.org/299776@main">https://commits.webkit.org/299776@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/108a37c7bd5f4db873ddd25e20e68e5238eae118

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120108 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39802 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30453 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126453 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72181 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121984 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40498 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48379 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91216 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60523 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123060 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32351 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107693 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71767 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31385 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25799 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70080 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101830 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25986 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129361 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47028 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35675 "Found 1 new test failure: http/tests/websocket/tests/hybi/contentextensions/block-cookies-worker.py (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99838 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47394 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103881 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99681 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25312 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45133 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23155 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43659 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46890 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52596 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46358 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49705 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48042 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/html/HTMLMarqueeElement.cpp:
```diff
@@ -119,7 +119,7 @@ void HTMLMarqueeElement::collectPresentationalHintsForAttribute(const QualifiedN
     case AttributeNames::scrolldelayAttr:
         if (!value.isEmpty())
-            addPropertyToPresentationalHintStyle(style, CSSPropertyWebkitMarqueeSpeed, limitToOnlyHTMLNonNegative(value, RenderStyle::initialMarqueeSpeed()), CSSUnitType::CSS_MS);
+            addPropertyToPresentationalHintStyle(style, CSSPropertyWebkitMarqueeSpeed, limitToOnlyHTMLNonNegative(value, 85), CSSUnitType::CSS_MS);
```
Surprised to see a literal number here rather than a named constant. Often we hoist such values to the top of the file?

weinig replied:
Yeah, good point. Eventually they can probably just be in the IDL as that is where they are defined:

https://html.spec.whatwg.org/multipage/obsolete.html#the-marquee-element

darinadler commented on Source/WebCore/html/HTMLMarqueeElement.cpp:
```diff
@@ -157,24 +157,24 @@ void HTMLMarqueeElement::stop()
 unsigned HTMLMarqueeElement::scrollAmount() const
 {
-    return limitToOnlyHTMLNonNegative(attributeWithoutSynchronization(scrollamountAttr), RenderStyle::initialMarqueeIncrement().intValue());
+    return limitToOnlyHTMLNonNegative(attributeWithoutSynchronization(scrollamountAttr), 6);
```
Surprised to see a literal number here rather than a named constant. Often we hoist such values to the top of the file?

darinadler commented on Source/WebCore/html/HTMLMarqueeElement.cpp:
```diff
@@ -157,24 +157,24 @@ void HTMLMarqueeElement::stop()
 void HTMLMarqueeElement::setScrollAmount(unsigned scrollAmount)
 {
-    setUnsignedIntegralAttribute(scrollamountAttr, limitToOnlyHTMLNonNegative(scrollAmount, RenderStyle::initialMarqueeIncrement().intValue()));
+    setUnsignedIntegralAttribute(scrollamountAttr, limitToOnlyHTMLNonNegative(scrollAmount, 6));
```
Surprised to see a literal number here rather than a named constant. Often we hoist such values to the top of the file?

darinadler commented on Source/WebCore/html/HTMLMarqueeElement.cpp:
```diff
@@ -157,24 +157,24 @@ void HTMLMarqueeElement::stop()
 unsigned HTMLMarqueeElement::scrollDelay() const
 {
-    return limitToOnlyHTMLNonNegative(attributeWithoutSynchronization(scrolldelayAttr), RenderStyle::initialMarqueeSpeed());
+    return limitToOnlyHTMLNonNegative(attributeWithoutSynchronization(scrolldelayAttr), 85);
```
Surprised to see a literal number here rather than a named constant. Often we hoist such values to the top of the file?

darinadler commented on Source/WebCore/html/HTMLMarqueeElement.cpp:
```diff
@@ -157,24 +157,24 @@ void HTMLMarqueeElement::stop()
 void HTMLMarqueeElement::setScrollDelay(unsigned scrollDelay)
 {
-    setUnsignedIntegralAttribute(scrolldelayAttr, limitToOnlyHTMLNonNegative(scrollDelay, RenderStyle::initialMarqueeSpeed()));
+    setUnsignedIntegralAttribute(scrolldelayAttr, limitToOnlyHTMLNonNegative(scrollDelay, 85));
```
Surprised to see a literal number here rather than a named constant. Often we hoist such values to the top of the file?

darinadler commented on Source/WebCore/rendering/RenderMarquee.cpp:
```diff
@@ -77,7 +77,7 @@ RenderMarquee::~RenderMarquee() = default;
 int RenderMarquee::marqueeSpeed() const
 {
-    int result = m_layer->renderer().style().marqueeSpeed();
+    auto result = static_cast<int>(m_layer->renderer().style().marqueeSpeed().asMilliseconds());
```
Is this the right kind of conversion? Flooring rather than rounding? Can it be out of `int` range?

weinig replied:
As per usual, it's the equivalent of the existing conversion, to keep behavior the same. Previously, this truncation happened when setting the value on RenderStyle. A better long term solution is probably to just keep it a double here.

weinig committed:
[Style] Convert the internal marquee properties to strong style types
https://bugs.webkit.org/show_bug.cgi?id=298475

Reviewed by Darin Adler.

Converts the internal marquee properties, `-webkit-marquee-increment`,
`-webkit-marquee-repetition` and `-webkit-marquee-speed`, to strong
style types.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/html/HTMLMarqueeElement.cpp:
* Source/WebCore/rendering/RenderMarquee.cpp:
* Source/WebCore/rendering/RenderMarquee.h:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleMarqueeData.cpp:
* Source/WebCore/rendering/style/StyleMarqueeData.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/values/non-standard/StyleWebKitMarqueeIncrement.h: Added.
* Source/WebCore/style/values/non-standard/StyleWebKitMarqueeRepetition.h: Added.
* Source/WebCore/style/values/non-standard/StyleWebKitMarqueeSpeed.h: Added.

Canonical link: https://commits.webkit.org/299776@main
